integration-tests/opentelemetry-reactive-messaging/src/main/java/io/quarkus/it/opentelemetry/TracedService.java

```diff
@@ -0,0 +1,13 @@
+package io.quarkus.it.opentelemetry;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.opentelemetry.extension.annotations.WithSpan;
+
+@ApplicationScoped
+public class TracedService {
+    @WithSpan
```

Comment:
But I have other concerns. It would be preferable to create an instrumenter for Kafka messages that does this tracing automatically without the need for the interceptor to kick in.

Reply:
This span is not necessary for the Kafka message tracing to work. It is there to test if propagation does work with CDI spans. The instrumenter for Kafka is already on Reactive Messaging side. Am I missing something?

Reply:
This is per messaging backend, for example for incoming messages : 

- Kafka : https://github.com/ozangunalp/smallrye-reactive-messaging/blob/db926afb1578d397a00a44dc31949b281b963d28/smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java#L247
- AMQP : https://github.com/ozangunalp/smallrye-reactive-messaging/blob/db926afb1578d397a00a44dc31949b281b963d28/smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java#L442
- RabbitMQ : https://github.com/ozangunalp/smallrye-reactive-messaging/blob/db926afb1578d397a00a44dc31949b281b963d28/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/tracing/TracingUtils.java#L79

Reply:
got it.